xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/shortcuts.vm

```diff
@@ -131,16 +131,16 @@
         req.send(null);
     };
 
-    // Append developer shortcuts for toggeling userType and hiddenDocuments in the current user profile
-    shortcut.add("$services.localization.render('core.shortcuts.developer.user.type')", function() {
+    // Append developer shortcuts for toggling userType and hiddenDocuments in the current user profile
+    shortcut.add("#getShortcutValue('core.shortcuts.developer.user.type')", function() {
         developerShortcutsRestCall("${request.contextPath}/rest/currentuser/properties/usertype/next",
                 "$escapetool.javascript($services.localization.render('core.shortcuts.developer.user.type.error'))");
-    }, {'type': shortcut.type.SEQUENCE, 'disable_in_input': true });
+    }, {'type': #if("$!xwiki.getDocument($xcontext.getUser()).getObject('XWiki.UserPreferenceShortcutsClass', 'translationKey', 'core.shortcuts.developer.user.type')"!='')shortcut.type.SIMPLE#{else}shortcut.type.SEQUENCE#end, 'disable_in_input': true });
```

Comment:
This call seems quite expensive. See also my comment about performance above.

Reply:
Addressed partially in https://github.com/xwiki/xwiki-platform/pull/2981/commits/bbc63a04652dd86aeec049ed67bb2336ade88bb1 
Does this look like a good enough solution?

Reply:
Yes, as written above, this looks okay from a performance point of view.